go/merkletree2/interfaces.go

```diff
@@ -0,0 +1,63 @@
+package merkletree2
+
+import "golang.org/x/net/context"
+
+// StorageEngine specifies how to store and lookup merkle tree nodes, roots and
+// KeyValuePairs. You can use a DB like Dynamo or SQL to do this.
+type StorageEngine interface {
+	// TODO update transaction management together with the blind architect
+	NewTransaction(context.Context) (Transaction, error)
+	CommitTransaction(context.Context, Transaction) error
+	AbortTransaction(context.Context, Transaction)
+
+	// StoreKVPairs stores the []KeyValuePair in the tree.
+	StoreKVPairs(context.Context, Transaction, Seqno, []KeyValuePair) error
+
+	StoreNode(context.Context, Transaction, Seqno, Position, Hash) error
+
+	StoreRootMetadata(context.Context, Transaction, RootMetadata) error
+
+	// LookupLatestRoot returns the latest root metadata and sequence number in
+	// the tree. If no root is found, then a NoLatestRootFound error is returned.
+	LookupLatestRoot(context.Context, Transaction) (Seqno, RootMetadata, error)
```

Comment:
I don't think `Transactions` should be in the `Lookup` signatures.  Also, I recommend that this interface has a function: `NewTreeBuilder` and an `type TreeBuilder interface` that should live only as long as one creation of a tree.  It can get the in-memory cache of the prior seqno's KV pairs from the `StorageEngine`.